src/include/kdbtypes.h

```diff
@@ -85,7 +89,7 @@ typedef unsigned char             kdb_octet_t;
 typedef signed short              kdb_short_t;
 typedef unsigned short            kdb_unsigned_short_t;
 
-#if SIZEOF_LONG == 4
+#if (LONG_MAX == +2147483647)
```

Comment:
If it runs into the ```# error``` we should add an additional apple branch and not messing up with existing typedefs.